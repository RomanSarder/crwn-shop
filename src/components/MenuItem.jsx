import React from 'react'
import styled from 'styled-components'

export const StyledMenuItem = styled.div`
    min-width: 30%;
    height: 24rem;
    flex: 1 1 auto;
    display: flex;
    align-items: center;
    justify-content: center;
    border: 1px solid black;
    margin: 0 0.75rem 1.5rem;

    .content {
        height: 9rem;
        padding: 0 2.5rem;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        border: 1px solid black;
    }

    .title {
        font-weight: bold;
        margin-bottom: 0.6rem;
        font-size: 2.2rem;
        color: #4a4a4a;
    }
`

export default function MenuItem({ title }) {

    return (
        <StyledMenuItem>
            <div className="content">
                <h1 className="title">{title}</h1>
                <span className="subtitle">SHOP NOW</span>
            </div>
        </StyledMenuItem>
    )
}
