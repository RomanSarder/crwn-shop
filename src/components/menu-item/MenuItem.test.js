import { render } from '@testing-library/react'
import 'jest-styled-components'
import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom'
import MenuItem from './MenuItem'
import userEvent from '@testing-library/user-event'

var history = createMemoryHistory()

var testLargeItem = {
    title: 'test item', 
    imageUrl: 'test-url', 
    size: 'large', 
    linkUrl: '/test-link-url'
}

var testSmallItem = {
    ...testLargeItem,
    size: undefined
}

function renderMenuItem (size = 'small') {
    var renderResult = render(
        <Router history={history}>
            { size == 'small' ? <MenuItem {...testSmallItem} /> : <MenuItem {...testLargeItem} /> }
        </Router>
    )

    return renderResult
}

it('should properly render small menu item', () => {
    var { container, getByText } = renderMenuItem()

    expect(container.firstChild).toMatchSnapshot()
    expect(getByText(testSmallItem.title)).toBeVisible()
    expect(getByText(/shop now/i)).toBeVisible()
})

it('should properly render large menu item', () => {
    var { container, getByText } = renderMenuItem('large')

    expect(container.firstChild).toMatchSnapshot()
    expect(getByText(testSmallItem.title)).toBeVisible()
    expect(getByText(/shop now/i)).toBeVisible()
})

it('should properly route to collection', () => {
    var { container } = renderMenuItem()

    userEvent.click(container.firstChild)

    expect(history.location.pathname).toEqual(testSmallItem.linkUrl)
})