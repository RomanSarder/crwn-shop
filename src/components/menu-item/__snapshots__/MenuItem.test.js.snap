// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should properly render large menu item 1`] = `
.c2 {
  height: 100%;
  width: 100%;
  z-index: -1;
  background-image: url(test-url);
  background-position: center;
  background-size: cover;
}

.c0 {
  position: relative;
  height: 24rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 2px solid #cac3c3;
  margin: 0 0.75rem 1.5rem;
  overflow: hidden;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover>.content {
  opacity: 0.9;
}

.c0:hover .c1 {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
  -webkit-transition: -webkit-transform 6s cubic-bezier(0.25,0.45,0.45,0.95);
  -webkit-transition: transform 6s cubic-bezier(0.25,0.45,0.45,0.95);
  transition: transform 6s cubic-bezier(0.25,0.45,0.45,0.95);
}

.c0 .content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 12rem;
  width: 16rem;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 2px solid black;
  background-color: white;
  opacity: 0.7;
  position: absolute;
}

.c0 .title {
  font-weight: bold;
  margin-bottom: 0.4rem;
  font-size: 2.4rem;
  color: #4a4a4a;
}

.c0 .subtitle {
  font-size: 1.8rem;
}

.c0 .title,
.c0 .subtitle {
  text-transform: uppercase;
}

@media (min-width:1200px) {
  .c0 {
    height: 38rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1 c2"
  />
  <div
    class="content"
  >
    <h1
      class="title"
    >
      test item
    </h1>
    <span
      class="subtitle"
    >
      SHOP NOW
    </span>
  </div>
</div>
`;

exports[`should properly render small menu item 1`] = `
.c2 {
  height: 100%;
  width: 100%;
  z-index: -1;
  background-image: url(test-url);
  background-position: center;
  background-size: cover;
}

.c0 {
  position: relative;
  height: 24rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 2px solid #cac3c3;
  margin: 0 0.75rem 1.5rem;
  overflow: hidden;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover>.content {
  opacity: 0.9;
}

.c0:hover .c1 {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
  -webkit-transition: -webkit-transform 6s cubic-bezier(0.25,0.45,0.45,0.95);
  -webkit-transition: transform 6s cubic-bezier(0.25,0.45,0.45,0.95);
  transition: transform 6s cubic-bezier(0.25,0.45,0.45,0.95);
}

.c0 .content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 12rem;
  width: 16rem;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 2px solid black;
  background-color: white;
  opacity: 0.7;
  position: absolute;
}

.c0 .title {
  font-weight: bold;
  margin-bottom: 0.4rem;
  font-size: 2.4rem;
  color: #4a4a4a;
}

.c0 .subtitle {
  font-size: 1.8rem;
}

.c0 .title,
.c0 .subtitle {
  text-transform: uppercase;
}

@media (min-width:1200px) {
  .c0 {
    height: 24rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1 c2"
  />
  <div
    class="content"
  >
    <h1
      class="title"
    >
      test item
    </h1>
    <span
      class="subtitle"
    >
      SHOP NOW
    </span>
  </div>
</div>
`;
